testutil-server/src/test/java/io/spine/testing/server/blackbox/BlackBoxBoundedContextTest.java

```diff
@@ -42,30 +46,96 @@
 import static io.spine.testing.server.blackbox.given.Given.addTask;
 import static io.spine.testing.server.blackbox.given.Given.createProject;
 import static io.spine.testing.server.blackbox.given.Given.createReport;
+import static io.spine.testing.server.blackbox.given.Given.createdProjectState;
 import static io.spine.testing.server.blackbox.given.Given.newProjectId;
 import static io.spine.testing.server.blackbox.given.Given.taskAdded;
+import static io.spine.testing.server.blackbox.verify.state.VerifyState.exactly;
+import static io.spine.testing.server.blackbox.verify.state.VerifyState.exactlyOne;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
+/**
+ * An abstract base for testing of black box bounded contexts.
+ */
 @ExtendWith(ShardingReset.class)
-@DisplayName("Black Box Bounded Context should")
-class BlackBoxBoundedContextTest {
+abstract class BlackBoxBoundedContextTest<T extends BlackBoxBoundedContext> {
 
-    private BlackBoxBoundedContext projects;
+    private T projects;
 
     @BeforeEach
     void setUp() {
-        projects = BlackBoxBoundedContext.newInstance()
-                                         .with(new BbProjectRepository());
+        projects = newInstance().with(new BbProjectRepository(),
+                                      new BbProjectViewRepository());
     }
 
     @AfterEach
     void tearDown() {
         projects.close();
     }
 
+    /**
+     * Creates a new instance of a bounded context for the basic set of tests.
```

Comment:
I am always confused by the "basic" term. Is there also an intermediate set of tests or an advanced one?